sdks/go/pkg/beam/core/runtime/exec/fn.go

```diff
@@ -261,21 +274,34 @@ func (n *invoker) ret2(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime,
 // ret3 handles processing of a trio of return values.
 func (n *invoker) ret3(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime, r0, r1, r2 interface{}) (*FullValue, error) {
 	switch {
-	case n.outErrIdx >= 0:
-		if r2 != nil {
-			return nil, r2.(error)
+	case n.outEtIdx >= 0:
+		if n.outErrIdx == 2 {
+			if r2 != nil {
+				return nil, r2.(error)
+			}
+			n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn}
+			return &n.ret, nil
 		}
-		if n.outEtIdx < 0 {
-			n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Elm2: r1, Pane: pn}
+		if n.outPcIdx >= 0 {
+			n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn, Continuation: r2.(sdf.ProcessContinuation)}
 			return &n.ret, nil
 		}
-		n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Pane: pn}
-		return &n.ret, nil
-	case n.outEtIdx == 0:
 		n.ret = FullValue{Windows: ws, Timestamp: r0.(typex.EventTime), Elm: r1, Elm2: r2, Pane: pn}
 		return &n.ret, nil
 	default:
-		panic(fmt.Sprintf("invoker.ret3: %T, %T, and %T don't match permitted return values.", r0, r1, r2))
+		if n.outErrIdx == 2 {
```

Comment:
I think this reads cleaner if you move it up to a condition in the outer switch:
```
switch {
case n.outEtIdx >= 0:
   // Do stuff
case n.outErrIdx == 2:
   // Do stuff
default:
   n.ret = FullValue{Windows: ws, Timestamp: ts, Elm: r0, Elm2: r1, Pane: pn, Continuation: r2.(sdf.ProcessContinuation)}
   return &n.ret, nil

Reply:
If you don't do that, there's no reason to use a switch instead of a simple if/else here

Reply:
Good catch, changed

Reply:
Style wise (and compiler wise) they're identical so no change should occur from this comment.

The only exception for preferring an if-else ladder is that variables defined in earlier conditional clauses are in scope for the remaining branches eg:

```
if foo, ok := getFoo(); !ok {
  // ...
} else if bar := foo.DoBar(); bar.Frobbled() {
 //  ... can use foo and bar. ...
} else {
  // ... can use foo and bar. ...
}
```

Reply:
Are you asking for this to be changed back? Or left as-is?

Reply:
As I said...
> Style wise (and compiler wise) they're identical so no change should occur from this comment.

It's also my job to tell you these sorts of things about Go, and the opportunity is during relevant context, even if it's not changing the code.

Reply:
Ah I misunderstood the statement, I interpreted it as "Danny's comment should not have resulted in a change"